benchmark/infrastructure/Benchmark.h

```diff
@@ -11,55 +11,56 @@
 #include "util/json.h"
 #include "util/Timer.h"
 #include "util/HashMap.h"
 #include "util/Exception.h"
+#include "util/SourceLocation.h"
 #include "../benchmark/util/HashMapWithInsertionOrder.h"
-#include "../benchmark/BenchmarkMetadata.h"
-#include "../benchmark/BenchmarkMeasurementContainer.h"
+#include "../benchmark/infrastructure/BenchmarkMetadata.h"
+#include "../benchmark/infrastructure/BenchmarkMeasurementContainer.h"
+#include "util/CopyableUniquePtr.h"
 
+namespace ad_benchmark{
 /*
  * Used for measuring the time needed for the execution of a function and
  * organizing those measured times.
  */
-class BenchmarkRecords {
+class BenchmarkResults {
     template<typename T>
-    using PointerVector = std::vector<std::unique_ptr<T>>;
+    using PointerVector = std::vector<ad_utility::CopyableUniquePtr<T>>;
 
     // A vector of all the created single measurements.
-    PointerVector<BenchmarkMeasurementContainer::RecordEntry>
-    singleMeasurements_;
+    PointerVector<ResultEntry> singleMeasurements_;
 
-    // A vector of all the created RecordGroups.
-    PointerVector<BenchmarkMeasurementContainer::RecordGroup>
-    recordGroups_;
+    // A vector of all the created resultGroups.
+    PointerVector<ResultGroup> resultGroups_;
   
-    // A hash map of all the created RecordTables. For faster access.
-    // The key for a RecordTable is it's descriptor.
-    PointerVector<BenchmarkMeasurementContainer::RecordTable>
-    recordTables_;
+    // A hash map of all the created resultTables. For faster access.
+    // The key for a resultTable is it's descriptor.
+    PointerVector<ResultTable> resultTables_;
 
     /*
     @brief Adds an entry to the given vector, by creating an instance of
-    `std::unique_ptr` for the given type and appending it. Strictly an
+    `CopyableUniquePtr` for the given type and appending it. Strictly an
     internal helper function.
 
-    @tparam EntryType The type, that the `std::unique_ptr` in the vector points
+    @tparam EntryType The type, that the `CopyableUniquePtr` in the vector points
     to.
     @tparam ConstructorArgs Types for the constructor arguments for creating
     a new instance of `EntryType`.
 
-    @param targetVector A vector of `std::unique_ptr`, that own values.
+    @param targetVector A vector of `CopyableUniquePtr`, that own values.
     @param constructorArgs Arguments to pass to the constructor of the object,
-    that the new `std::unique_ptr` will own.
+    that the new `CopyableUniquePtr` will own.
     */
     template<typename EntryType, typename... ConstructorArgs>
     static EntryType& addEntryToContainerVector(
       PointerVector<EntryType>& targetVector,
       ConstructorArgs&&... constructorArgs){
-      targetVector.push_back(std::make_unique<EntryType>(
+      targetVector.push_back(
+        ad_utility::make_copyable_unique<EntryType>(
         std::forward<ConstructorArgs>(constructorArgs)...));
```

Comment:
```suggestion
      targetVector.push_back(
        ad_utility::make_copyable_unique<EntryType>(
        AD_FWD(constructorArgs)...));
```
(in `util/Forward.h`)

Reply:
Done, see [95d60a0](https://github.com/ad-freiburg/qlever/pull/860/commits/95d60a0f0a0d90f44db4e0d9c433f1e4a8bf86aa).